include/sbi/sbi_scratch.h

```diff
@@ -57,6 +58,7 @@ struct sbi_scratch {
 	unsigned long hartid_to_scratch;
 	/** IPI type (or flags) */
 	unsigned long ipi_type;
+	unsigned long next_arg2;
 } __packed;
```

Comment:
Rename next_arg2 to prev_arg1. It is good to save Arg1 (a1 register) passed by previous booting stage in per-HART scratch space.

Also, define prev_arg1 before "next_arg1".

Reply:
If we don't have built-in DTB linked to OpenSBI then prev_arg1 will point to "a1" register passed by previous booting stage.